src/SignalR/perf/benchmarkapps/BenchmarkServer/Hubs/EchoHub.cs

```diff
@@ -30,6 +35,29 @@ public async Task Broadcast(int duration)
             Console.WriteLine("Broadcast exited: Sent {0} messages", sent);
         }
 
+        public static string Status => $"{_connectionCount} current, {_peakConnectionCount} peak.";
+
+        public void LogConnections(string label) {
+            if (_connectionCount < 1000 || _connectionCount % 100 == 0)
```

Comment:
Do you find this a bit noisy for the first 1000 connections?

Reply:
I did this so I can get an earlier understanding of how quickly multiple workers (and clients) are connecting, including if any are yet to connect.

That said, after 100-200 that early understanding has been received and the rest up until 1000 is just noise.  I'll drop it down to 100.

Reply:
Before the if you should use Interlocked.Read to get connectionCount locally without tearing:

```
var connectionCount = Interlocked.Read(ref _connectionCount);
```

https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.read?view=netcore-3.0

Reply:
switched for `EchoHubConnectionCounter`